### DISCUSSION TITLE
Question about security/overwrite of sudo

### DISCUSSION THREAD

**Discussion by SESAgger** [Homebrew/discussions#2974] [Everyday usage]
So, I just came across this [article](https://infotoast.org/site/index.php/2021/05/30/homebrew-is-a-major-security-flaw-install-macports-instead/) that claims that homebrew opens you up to having your password grabbed.
To me that just sounds completely insane that apple would allow that to be possible, so is it true and if yes, is there anything to be done to fix it?
I'm pretty hopeless with these things so bear over with me if the summary is bad.

**Comment by SMillerDev** [comment]
That blogpost misses the point of a compromised system in favor of bashing on Homebrew.

> Imagine if someone wanted to make a fake sudo command that grabs your password to elevate itself to administrator. In Macs that have Homebrew installed, as the sudo binary is located in /usr/bin, a malicious program can place a script called “sudo” in /usr/local/bin or /opt/local/bin that takes your password and then uses it to allow a virus to gain administrator access.

If someone has this kind of access to your machine they can also edit your PATH to include `/some/evil/path/` where they place the `sudo` binary that grabs your password. Homebrew being installed somewhere doesn't have any effect on this.

The article additionally makes claims that this same thing applies to Homebrew on Apple Silicon, which is false.

> Homebrew made the installation directory for packages, /usr/local/bin (Yes, I know it’s /opt/local/bin on Apple Silicon)

Except it's `/opt/homebrew/bin` on Apple silicon

> /opt/local/bin <- Homebrew makes this writable by you on Apple Silicon

Homebrew doesn't, again it's `/opt/homebrew/bin` and `/opt/local/bin` isn't in the default path from Apple because it's nonsensical in a Unix filesystem context.

> If you’re on Apple Silicon, you will still need to remove the /opt/homebrew directory from your system.

This is the only time the post gets the Apple Silicon Homebrew path correct.

**Comment by carlocab** [comment]
Using a package manager like Homebrew instead of MacPorts (or vice-versa) just changes the attack surface -- whether the attack surface is larger or smaller depends largely how you use your machine.

If you use `brew`, you do become marginally more vulnerable to having a malicious `sudo` placed on your `PATH`, but if an attacker can do that, they likely already have other ways to compromise your machine (see @SMillerDev's answer above). That said, if you use `brew`, you rarely ever have to use `sudo`, so you're also less vulnerable to being exploited by a malicious executable masquerading as `sudo`.

If you use MacPorts, then you have the reverse of the above. You are marginally less vulnerable to having a malicious `sudo` placed on your `PATH`. On the other hand, you need to use `sudo` _all the time_ (you need `sudo` to be able to install anything with MacPorts). This makes you a lot more vulnerable to exploitation by an attacker trying to highjack `sudo`.
